File: test/nodejs/duckdb_pgwire_test.js
var duckdb = require('../../duckdb/tools/nodejs');
var assert = require('assert');

describe(`duckdb_pgwire extension`, () => {
    let db;
    let conn;
    before((done) => {
        db = new duckdb.Database(':memory:', {"allow_unsigned_extensions":"true"});
        conn = new duckdb.Connection(db);
        conn.exec(`LOAD '${process.env.DUCKDB_PGWIRE_EXTENSION_BINARY_PATH}';`, function (err) {
            if (err) throw err;
            done();
        });
    });

    it('duckdb_pgwire function should return expected string', function (done) {
        db.all("SELECT duckdb_pgwire('Sam') as value;", function (err, res) {
            if (err) throw err;
            assert.deepEqual(res, [{value: "DuckdbPgwire Sam 🐥"}]);
            done();
        });
    });

    it('duckdb_pgwire_openssl_version function should return expected string', function (done) {
        db.all("SELECT duckdb_pgwire_openssl_version('Michael') as value;", function (err, res) {
            if (err) throw err;
            assert(res[0].value.startsWith('DuckdbPgwire Michael, my linked OpenSSL version is OpenSSL'));
            done();
        });
    });
});

File: test/python/duckdb_pgwire_test.py
import duckdb
import os
import pytest

# Get a fresh connection to DuckDB with the duckdb_pgwire extension binary loaded
@pytest.fixture
def duckdb_conn():
    extension_binary = os.getenv('DUCKDB_PGWIRE_EXTENSION_BINARY_PATH')
    if (extension_binary == ''):
        raise Exception('Please make sure the `DUCKDB_PGWIRE_EXTENSION_BINARY_PATH` is set to run the python tests')
    conn = duckdb.connect('', config={'allow_unsigned_extensions': 'true'})
    conn.execute(f"load '{extension_binary}'")
    return conn

def test_duckdb_pgwire(duckdb_conn):
    duckdb_conn.execute("SELECT duckdb_pgwire('Sam') as value;");
    res = duckdb_conn.fetchall()
    assert(res[0][0] == "DuckdbPgwire Sam 🐥");

def test_duckdb_pgwire_openssl_version_test(duckdb_conn):
    duckdb_conn.execute("SELECT duckdb_pgwire_openssl_version('Michael');");
    res = duckdb_conn.fetchall()
    assert(res[0][0][0:51] == "DuckdbPgwire Michael, my linked OpenSSL version is OpenSSL");